server/service/src/document/update_trigger.rs

```diff
@@ -0,0 +1,106 @@
+                Some(date_of_birth) => {
+                    Some(NaiveDate::from_str(&date_of_birth).map_err(|err| {
+                        DocumentInsertError::InternalError(format!(
+                            "Invalid date of birth format: {}",
+                            err
+                        ))
+                    })?)
+                }
+                None => None,
+            };
+            let address = patient.addresses.get(0);
+            let name_repo = NameRowRepository::new(con);
+            let patient_name = |first: &Option<String>, last: &Option<String>| {
+                let mut out = vec![];
+                if let Some(first) = first {
+                    out.push(first.clone());
+                }
+                if let Some(last) = last {
+                    out.push(last.clone());
+                }
+                out.join(",")
+            };
+            let existing_name = name_repo.find_one_by_id(&patient.id)?;
+            name_repo.upsert_one(&NameRow {
+                id: patient.id.clone(),
+                name: patient_name(&patient.first_name, &patient.last_name),
+                code: "program_patient".to_string(),
+                r#type: NameType::Patient,
+                is_customer: true,
+                is_supplier: false,
+                supplying_store_id: Some(store_id.to_string()),
+                first_name: patient.first_name,
+                last_name: patient.last_name,
+                gender: patient.gender.and_then(|g| match g {
+                    Gender::Female => Some(NameRowGender::Female),
+                    Gender::Male => Some(NameRowGender::Male),
+                    Gender::TransgenderMale => Some(NameRowGender::TransgenderMale),
+                    Gender::TransgenderFemale => Some(NameRowGender::TransgenderFemale),
+                    Gender::Unknown => Some(NameRowGender::Unknown),
+                    Gender::NonBinary => Some(NameRowGender::NonBinary),
+                }),
+                date_of_birth,
+                charge_code: None,
+                comment: None,
+                country: address.and_then(|a| a.country.clone()),
+                address1: address.and_then(|a| a.address_1.clone()),
+                address2: address.and_then(|a| a.address_2.clone()),
+                phone: contact.and_then(|c| c.phone.clone().or(c.mobile.clone())),
+                email: contact.and_then(|c| c.email.clone()),
+                website: contact.and_then(|c| c.website.clone()),
+                is_manufacturer: false,
+                is_donor: false,
+                on_hold: false,
+                created_datetime: existing_name
+                    .as_ref()
+                    .and_then(|n| n.created_datetime.clone())
+                    .or(Some(doc.timestamp.naive_utc())), // assume there is no earlier doc version
+            })?;
+            if existing_name.is_none() {
```

Comment:
I think checking against name_store_join is safer, and don't need existing_name.as_ref() if we do it that way.

Reply:
done somewhere...